### ISSUE TITLE
First load of codesanbox when logged in bug

### ISSUE THREAD

**Issue by coolemur** [codesandbox/codesandbox-client#6032]
# 🐛 bug report

## Description of the problem

I have a simple fetch in vanilla ts codesanbox, which loads some data from api and displays it on page.

```
//index.ts

async function fetchData(url: string) {
  const response = await fetch(url);
  return response.json();
}

fetchData("https://jsonplaceholder.typicode.com/todos").then((data) => {
  const app = document.getElementById("app");

  if (app)
    app.innerHTML = `
    <h1>Simple fetch example</h1><p><pre>${JSON.stringify(
      data,
      null,
      2
    )}</pre></p>`;
});

```

When I am logged in to my codesandbox account, on first load it doesn't work as expected.
It shows content for a moment and then clears it. If I'm not logged in, everything works as expected.

Issue #4128 might be related

### To Reproduce

Try to copy/paste code provided above to codesandbox and test first load while logged in.

### Your Environment

| Software         | Name/Version |
| ---------------- | ------------ |
| Сodesandbox      |       ?? - version should be easy to find on sanbox       |
| Browser          |       Chrome Version 92.0.4515.131 (Official Build) (x86_64)       |
| Operating System |       Mac OS Big Sur Version 11.2.3 (20D91)       |

**Comment by CompuIves** [comment]
Interesting, this sounds like a potential race condition. Can you see the issue happening here? https://codesandbox.io/s/elegant-rubin-w1cbq?file=/src/index.ts

**Comment by coolemur** [comment]
@CompuIves No, this one works fine, even when I'm logged in. (But the one that is in my account still doesn't work on first load.)

**Comment by lbogdan** [comment]
I can consistently reproduce this with a sandbox that I own (https://codesandbox.io/s/issue-6032-zi0di), while it never happens when logged out, or with a sandbox I don't own.

@DeMoorJasper do you have any idea why this might happen?

**Comment by DeMoorJasper** [comment]
@lbogdan I think this is due to the preview re-rendering when the live features are loaded, this PR would probably fix it: #5680 

Another fix would probably be ensuring our html hmr renders everything before we execute the code, although a better solution for this would probably be refreshing the page on each html change.
